mutable_tree.go

```diff
@@ -117,33 +117,33 @@ func (tree *MutableTree) String() (string, error) {
 
 // Set/Remove will orphan at most tree.Height nodes,
 // balancing the tree after a Set/Remove will orphan at most 3 nodes.
-func (tree *MutableTree) prepareOrphansSlice() []*Node {
-	return make([]*Node, 0, tree.Height()+3)
+func (tree *MutableTree) prepareOrphansSlice() []ComplexNode {
```

Comment:
1. Can we make this a `Node` interface?
2. Orphans do not need to store that much data, only need hash and a version (see `orphansKeyFormat` in `nodeDb`). Can we make a separate struct for them?